framework/src/meshgenerators/MeshDiagnosticsGenerator.C

```diff
@@ -335,6 +343,66 @@ MeshDiagnosticsGenerator::checkWaterTightSidesets(const std::unique_ptr<MeshBase
   diagnosticsLog(message, _check_watertight_sidesets, num_faces_without_sideset);
 }
 
+void
+MeshDiagnosticsGenerator::checkWatertightNodesets(const std::unique_ptr<MeshBase> & mesh) const
+{
+  /*
+  Diagnostic Overview:
+  1) Mesh precheck
+  2) Loop through all elements
+  3) Loop through all sides of that element
+  4) If side is external loop through its nodes
+  5) If node is not associated with any nodeset add to list
+  6) Print out node id
+  */
+  if (mesh->mesh_dimension() < 2)
+    mooseError("The nodeset check only works for 2D and 3D meshes");
+  auto & boundary_info = mesh->get_boundary_info();
+  boundary_info.build_side_list();
+  const auto nodeset_map = boundary_info.get_nodeset_map();
+  unsigned int num_nodes_without_nodeset = 0;
+  std::set<dof_id_type> checked_nodes_id;
+
+  for (const auto elem : mesh->active_element_ptr_range())
+  {
+    for (const auto i : elem->side_index_range())
+    {
+      // Check if side is external
+      if (elem->neighbor_ptr(i) == nullptr)
+      {
+        // Side is external, now check nodes
+        auto side = elem->side_ptr(i);
+        const auto & node_list = side->get_nodes();
+        for (unsigned int j = 0; j < side->n_nodes(); j++)
+        {
+          const auto node = node_list[j];
+          if (checked_nodes_id.count(node->id()))
+            continue;
+          // if node is not part of nodeset map add it to list of bad nodes
+          if (nodeset_map.count(node) == 0)
```

Comment:
Could we just test `boundary_info.n_boundary_ids(node)` here?  The raw `get_nodeset_map()` accessor was an encapsulation mistake I should never have approved, and I still vaguely dream of deprecating it someday.

Reply:
Just want to check, would that return true if the node is assigned to a sideset but not a nodeset?